### PR TITLE
EVG-14178 Log cannot map pull request to project at a lower level

### PR DESCRIPTION
**Pull request by bynn**
[EVG-14178](https://jira.mongodb.org/browse/EVG-14178)

### Description 
Since `FindOneProjectRefByRepoAndBranchWithPRTesting` already logs a debug message, I just deleted it

### Testing 
n\a

### CODE REVIEW

File: rest/data/patch_intent.go [hunk 1]
```diff
@@ -19,20 +19,14 @@ type DBPatchIntentConnector struct{}
...
+	_, err := model.FindOneProjectRefByRepoAndBranchWithPRTesting(patchDoc.GithubPatchData.BaseOwner,
 		patchDoc.GithubPatchData.BaseRepo, patchDoc.GithubPatchData.BaseBranch)
 	if err != nil {
 		return gimlet.ErrorResponse{
 			StatusCode: http.StatusInternalServerError,
 			Message:    "failed to fetch project_ref",
 		}
 	}
-	if projectRef == nil {
-		return gimlet.ErrorResponse{
```
Comment (ybrill) [inline rest/data/patch_intent.go:31]: I think it's still valuable to return here. For example, if the project has PR testing disabled we don't want to continue to create the patch.

Comment (bynn) [reply rest/data/patch_intent.go:31]: returns nil

Comment (ablack12) [reply rest/data/patch_intent.go:31]: I think we should still log this to grip; it's possible that we will want this information if users ask why a PR isn't being made, but logging at the Debug level might be more accurate (and then returning nil).

Comment (ablack12) [reply rest/data/patch_intent.go:31]: Oh just saw your comment that we already log at the debug level so disregard.

**Review by ybrill — APPROVED** [review]

**Pull request merged**